python/ray/data/_internal/block_batching/iter_batches.py

```diff
@@ -36,6 +40,7 @@ def iter_batches(
     shuffle_seed: Optional[int] = None,
     ensure_copy: bool = False,
     prefetch_batches: int = 1,
+    dataset_tag: str = "unknown_dataset",
```

Comment:
can we make this arg required? I'm concerned that we'll miss it for some places.